bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseAudioAudioSource.java

```diff
@@ -133,24 +135,46 @@ private synchronized void registerPipe(PipedOutputStream pipeOutput) {
         startPipeWrite();
     }
 
-    private void startPipeWrite() {
-        if (pipeWriteTask == null) {
+    private synchronized void startPipeWrite() {
+        if (this.pipeWriteTask == null) {
             this.pipeWriteTask = executor.submit(() -> {
                 int lengthRead;
                 byte[] buffer = new byte[1024];
+                int readRetries = 3;
                 while (!pipeOutputs.isEmpty()) {
                     var stream = getSourceInputStream();
                     if (stream != null) {
                         try {
                             lengthRead = stream.read(buffer);
+                            readRetries = 3;
                             for (var output : pipeOutputs) {
-                                output.write(buffer, 0, lengthRead);
-                                output.flush();
+                                try {
+                                    output.write(buffer, 0, lengthRead);
+                                    if (pipeOutputs.contains(output)) {
+                                        output.flush();
+                                    }
+                                } catch (IOException e) {
+                                    if (e instanceof InterruptedIOException && pipeOutputs.isEmpty()) {
```

Comment:
Is there a hierarchy link between InterruptedIOException and IOException? 
Why not catching InterruptedIOException directly?

Reply:
Verified. No link. 
Your condition will never be true.